hls4ml/model/optimizer/passes/bn_quant.py

```diff
@@ -16,35 +16,12 @@ def initialize(self):
         inp = self.get_input_variable()
         shape = inp.shape
         dims = inp.dim_names
-        precision = self.model.config.backend.convert_precision_string(inp.type.precision)
-        W, I, F = precision.width, precision.integer, precision.fractional
-        original_name = self.attributes.get('original_name')
-        variance = self.model.get_weights_data(original_name, 'moving_variance')
-        mean = self.model.get_weights_data(original_name, 'moving_mean')
-        gamma = self.model.get_weights_data(original_name, 'gamma')
-        beta = self.model.get_weights_data(original_name, 'beta')
```

Comment:
Moved the setting of the thresholds from the layer `initialize` to a separate function because the `BatchNormalization` layer from which the thresholds are derived may have been inserted by another optimization pass (`QKerasFactorizeAlpha` or new `ExtractTernaryThreshold`). In that case the layer doesn't exist in the original model and , e.g., `variance = self.model.get_weights_data(original_name, 'moving_variance')` won't work